solana/pyth2wormhole/client/src/attestation_cfg.rs

```diff
@@ -41,66 +33,25 @@ pub struct AttestationConfig {
+    /// Attestation conditions that will be used for any symbols included in the mapping
+    /// that aren't explicitly in one of the groups below.
+    pub default_attestation_conditions: AttestationConditions,
+
+    /// Collection of symbols identified by symbol name (e.g., "Crypto.BTC/USD")
+    /// These symbols are only active if `mapping_addr` is set.
+    pub name_groups: Vec<NameGroup>,
+
+    /// Collection of symbols identified by their full account addresses.
+    /// These symbols will be published regardless of whether or not `mapping_addr` is provided.
+    pub symbol_groups: Vec<SymbolGroup>,
 }
 
 impl AttestationConfig {
-    /// Merges new symbols into the attestation config. Pre-existing
-    /// new symbols are ignored. The new_group_name group can already
-    /// exist - symbols will be appended to `symbols` field.
-    pub fn add_symbols(
-        &mut self,
-        mut new_symbols: HashMap<Pubkey, HashSet<Pubkey>>,
-        group_name: String, // Which group is extended by the new symbols
-    ) {
-        // Remove pre-existing symbols from the new symbols collection
-        for existing_group in &self.symbol_groups {
-            for existing_sym in &existing_group.symbols {
-                // Check if new symbols mention this product
-                if let Some(prices) = new_symbols.get_mut(&existing_sym.product_addr) {
-                    // Prune the price if exists
-                    prices.remove(&existing_sym.price_addr);
-                }
-            }
-        }
-
-        // Turn the pruned symbols into P2WSymbol structs
-        let mut new_symbols_vec = new_symbols
-            .drain() // Makes us own the elements and lets us move them
-            .flat_map(|(prod, prices)| iter::zip(iter::repeat(prod), prices)) // Flatten the tuple iterators
-            .map(|(prod, price)| P2WSymbol {
-                name:         None,
-                product_addr: prod,
-                price_addr:   price,
-            })
-            .collect::<Vec<P2WSymbol>>();
-
-        // Find and extend OR create the group of specified name
-        match self
-            .symbol_groups
-            .iter_mut()
-            .find(|g| g.group_name == group_name) // Advances the iterator and returns Some(item) on first hit
-        {
-            Some(existing_group) => existing_group.symbols.append(&mut new_symbols_vec),
-            None if !new_symbols_vec.is_empty() => {
-                // Group does not exist, assume defaults
-                let new_group = SymbolGroup {
-                    group_name,
-                    conditions: Default::default(),
-                    symbols: new_symbols_vec,
-                };
-
-                self.symbol_groups.push(new_group);
-            }
-            None => {}
-        }
-    }
-
-    pub fn as_batches(&self, max_batch_size: usize) -> Vec<BatchState> {
+    pub fn as_batches(&self, max_batch_size: usize) -> Vec<SymbolGroup> {
```

Comment:
I think this method and attestation_config_to_batches should live in the same place (or maybe even in one function) and probably as `AttestationConfig` implementation.